scripts/create-container-registry-creds.sh

```diff
@@ -4,6 +4,7 @@ set -aueo pipefail
 
 # shellcheck disable=SC1091
 source .env
+USE_PRIVATE_REGISTRY="${USE_PRIVATE_REGISTRY:-false}"
```

Comment:
Why is `USE_PRIVATE_REGISTRY` redefined here? If I wanted to change the env var for the osm multicluster demo I would have to update it in  `create-container-registry-creds.sh`, `demo/run-osm-multicluster-demo.sh` and `demo/configure-app-namespaces.sh`.

Reply:
I set it in `demo/run-osm-demo.sh` in a separate PR: https://github.com/openservicemesh/osm/pull/3872. I didn't want to change too many files in that PR so that's why I'm making these changes here. 

I just redefined it in this file here in case someone wants to run this script by itself, or in case there's a job / demo that runs the script without setting USE_PRIVATE_REGISTRY.

Reply:
The expectation here is that users set their .env files for the non-automated-demo scripts. 

See discussion here: https://github.com/openservicemesh/osm/pull/3839#discussion_r675160685

But I agree it is confusing to users since I don't think it is documented that the defaults only apply to the demo scripts and not each individual script. However we don't want to keep track of maintaining the defaults in a bunch of different locations

Reply:
Got it, thanks for linking the discussion.